pkg/http/client.go

```diff
@@ -0,0 +1,136 @@
+package http
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/dustin/go-humanize"
+
+	"github.com/kudobuilder/kuttl/pkg/version"
+)
+
+// Client is client used to simplified http requests for tarballs
+type Client struct {
+	client    *http.Client
+	UserAgent string
+}
+
+// Get performs an HTTP get and returns the Response.  The caller is responsible for closing the response.
+func (c *Client) Get(url string) (*http.Response, error) {
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("User-Agent", c.UserAgent)
+
+	resp, err := c.client.Do(req)
+	return resp, err
+}
+
+// Get performs HTTP get, retrieves the contents and returns a bytes.Buffer of the entire contents
+// this could be dangerous against an extremely large file
+func (c *Client) GetByteBuffer(url string) (*bytes.Buffer, error) {
+	buf := bytes.NewBuffer(nil)
+
+	resp, err := c.Get(url)
+	if err != nil {
+		return buf, err
+	}
+	if resp.StatusCode != 200 {
+		return buf, fmt.Errorf("failed to fetch %s : %s", url, resp.Status)
+	}
+
+	_, err = io.Copy(buf, resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	err = resp.Body.Close()
+	if err != nil {
+		return nil, fmt.Errorf("error when closing the response body %s", err)
+	}
+	return buf, err
+}
+
+// DownloadFile expects a url with a file and will save that file to the path provided preserving the file name.
+func (c *Client) DownloadFile(url, path string) (string, error) {
+	filePath := filepath.Join(path, filepath.Base(url))
+	return filePath, c.Download(url, filePath)
+}
+
+// Download takes a url to download and a filepath to write it to
+// this will write the response to any url request to a file.
+func (c *Client) Download(url string, path string) error {
+	// Get the data
+	resp, err := c.Get(url)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
```

Comment:
I guess we should have a check if the file already exists here and error out if the file exists?